skbio/metadata/_feature.py

```diff
@@ -50,20 +50,32 @@ def __hash__(self):
             self._hash = hash(frozenset(self.items()))
         return self._hash
 
-    def __lt__(self, other):
-        return self['location'] < other['location']
+    def __eq__(self, other):
+        return hash(self) == hash(other)
 
-    def __le__(self, other):
-        return self['location'] <= other['location']
+    def __ne__(self, other):
+        return hash(self) != hash(other)
```

Comment:
how about ` return not self.__eq__(other)`? minor thing to further decrease redundancy